full-stack-practice/client/src/lib/read.ts

```diff
@@ -0,0 +1,24 @@
+import { type Product } from './data';
+
+// Fetch all products
+export async function readCatalog(): Promise<Product[]> {
+  const response = await fetch('/api/products');
+  if (!response.ok) {
+    throw new Error(`Failed to fetch products. Status: ${response.status}`);
+  }
+  return response.json();
+}
+
+// Fetch a single product by ID
+export async function readProduct(
+  productId: number
+): Promise<Product | undefined> {
+  const response = await fetch(`/api/products/${productId}`);
+  if (!response.ok) {
+    if (response.status === 404) {
```

Comment:
I'm not sure why you have this special check, but it doesn't hurt anything to do it this way.